src/main/webapp/event.js

```diff
@@ -364,20 +376,56 @@ class Event {
       AMOrPM = "PM";
       hours -= 12;
     } else if (hours == 12) {
-        AMOrPM ="PM";
+      AMOrPM = "PM";
     } else if (hours == 0) {
-        AMOrPM = "AM";
-        hours = 12;
+      AMOrPM = "AM";
+      hours = 12;
     } else {
       AMOrPM = "AM";
     }
-    
+
     let hoursString = hours.toString();
     let minutesString = minutes.toString();
-  
+
     if (minutes < 10) {
       minutesString = `0${minutes.toString()}`;
     }
     return `${hoursString}:${minutesString} ${AMOrPM}`;
   }
+
+  parseDate(dateAndHours) {
+    // Will change format from MM DD, YYYY 00:00AM to MM DD, YYYY
+    const date = dateAndHours.propertyMap.date.slice(0, -12);
+    return date;
+  }
+
+  parseDateForValue(dateAndHours) {
+    // Will translate a date in format MM DD, YYYY to YYYY-MM-DD for purpose of prefilling input date value
```

Comment:
Oof...  I would normally recommend using https://momentjs.com/  for this (and if you want to try, go for it!)

Reply:
I think I am going to keep it as is for now for the sake of time but I will definitely use it in the future!